driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -1134,6 +1148,20 @@ public Integer getSrvMaxHosts() {
         return srvMaxHosts;
     }
 
+    /**
+     * Gets the SRV service name according to RFC 6335, with the exception that it may exceed 15 characters as long as the 63rd (62nd with
+     * prepended underscore) character DNS query limit is not surpassed.
```

Comment:
I found where the 15 characters limit comes from (https://www.rfc-editor.org/rfc/rfc6335.html#section-5.1), but failed to find anything about the 62/63 character limit in both https://www.rfc-editor.org/rfc/rfc6335 and https://www.rfc-editor.org/rfc/rfc2782. Where is this limit specified? It may also be useful to express the limits as hyperlinks to the places where they are specified.

Another thing that confuses me is the "rd" and "nd" ordinal indicators. I read those as "sixty-thi**rd**" and "sixty-seco**nd**" ordinals, i.e., numbers that express position, not size/length/cardinality. And we need a cardinal number (not an ordinal one) when expressing the limit. Am I interpreting the "rd" and "nd" incorrectly here?

P.S. This comment is equally applicable to the corresponding methods in `ClusterSettings`.

Reply:
I see. It's here [here|https://github.com/stIncMale/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#srvservicename] in our spec, and the same "rd" and "nd" ordinal indicators are used there. Since you did not say anything about them, I deduce that you see no problem with the way the limit is expressed?

As for the actual source of the limit, I found that the [preferred name syntax](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.1) states that the name is a sequence of labels joined with `.`. Each label is limited to 63 characters, and the whole name is limited to 255 characters according to the [size limits](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.4)<sup>1</sup>. For an SRV record, the `_Service` part of the RR name is a label from the standpoint of the RR name, i.e., it is limited to 63 characters.

Now we have the links to the actual source of the limit, i.e., the situation has changed, and so I again propose to include the links in the docs to explain where the limit comes from.

---
<sup>1</sup> The RFC uses both terms "character" and "octet" when mentions the limit in different places, but given what characters are [allowed to be used as part of the name](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.1), they are all encoded with a single octet.

Reply:
I'm not seeing anything about "rd" or "nd" in the driver spec.  Where exactly are you seeing it?

I'm still unsure why we should say anything more than what's in the spec:

> This option specifies a valid SRV service name according to RFC 6335, with the exception that it may exceed 15 characters as long as the 63rd (62nd with prepended underscore) character DNS query limit is not surpassed. This option requires a string value and defaults to "mongodb". 

but if it's insufficient can you suggest the exact change you'd like to see?

Reply:
> I'm not seeing anything about "rd" or "nd" in the driver spec. Where exactly are you seeing it?

It's in _"63rd"_ and _"62nd"_. My interpretation and confusion are described in the second paragraph in the [first comment](https://github.com/mongodb/mongo-java-driver/pull/843#discussion_r773590353).

> why we should say anything more than what's in the spec

- At first, since we decided to specify the limits, I wanted to have links to the relevant part of relevant RFCs where the limits are specified.
- When I found those links (took time, and I don't want to burden users with the exercise, it's enough that our specification burdens us) and read parts of RFCs, I ended up having more questions:
  - Why do we suggest users to disregard the 15 characters limit (applicable to SRV records) and instead care about a more generic 63 characters limit (applicable to all DNS resource records)?
  - Why do we not mention the 255 characters limit on the whole DNS resource record name? Given that users supply both the service name and the host name parts of the SRV resource record name (specified by [RFC2782](https://www.rfc-editor.org/rfc/rfc2782)), this limit appears to be at least as relevant as the other two limits (arguably, more relevant than the 63 characters limit overridden by the 15 characters limit).

The links to all limits:
  - https://www.rfc-editor.org/rfc/rfc6335#section-5.1 for the 15 characters limit on the SRV service name;
  - https://www.rfc-editor.org/rfc/rfc1035#section-2.3.1 for the 63 characters limit on the label part of a resource record name (the SRV service name is a label in terms of this RFC);
  - https://www.rfc-editor.org/rfc/rfc1035#section-2.3.4 for the 255 characters limit on the whole resource record name.

> can you suggest the exact change you'd like to see?

For example:
The [SRV resource record](https://www.rfc-editor.org/rfc/rfc2782) ([RFC 2782](https://www.rfc-editor.org/rfc/rfc2782)) [service name, which is limited to 15 characters](https://www.rfc-editor.org/rfc/rfc6335#section-5.1) ([RFC 6335 section 5.1](https://www.rfc-editor.org/rfc/rfc6335#section-5.1)). If specified, it is combined with the single host name specified by [`ConnectionString.getHosts`](https://mongodb.github.io/mongo-java-driver/4.4/apidocs/mongodb-driver-core/com/mongodb/ConnectionString.html#getHosts()) as follows: `_srvServiceName._tcp.hostName`. The combined string is an SRV [resource record name](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.1) ([RFC 1035 section 2.3.1](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.1)), which is [limited to 255 characters](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.4) ([RFC 1035 section 2.3.4](https://www.rfc-editor.org/rfc/rfc1035#section-2.3.4)).

Reply:
Done.  I duplicated it in four places (getter and setter for MongoClientOptions/Settings) but not ConnectionString, where I just linked to the ClusterSettings docs.  That seems to fit the style better (brevity).

Reply:
Thanks 👍 The previous description is left on `MongoClientURI.getSrvServiceName`. If you remove it, there is no need to re-request review, the PR can then be merged right away.